packages/ra-core/src/sideEffect/recordForm.js

```diff
@@ -0,0 +1,36 @@
+import { put, takeEvery } from 'redux-saga/effects';
+import { LOCATION_CHANGE } from 'react-router-redux';
+import { resetForm } from '../actions/formActions';
+
+export const handleLocationChangeFactory = (
+    initialLocationPathname = undefined
+) => {
+    let lastLocationPathname = initialLocationPathname;
+
+    function* handleLocationChange({ payload: { pathname } }) {
+        // We don't want to reset the form when switching tabs in TabbedForm
+        // so we store the current location. If it was the first location change,
+        // we have nothing left to do
+        if (!lastLocationPathname) {
+            lastLocationPathname = pathname;
+            return;
+        }
+
+        // We check wether we are changing from a tab path to the form root path
+        // or vice-versa
+        if (
+            !lastLocationPathname.startsWith(pathname) &&
```

Comment:
I'm afraid of side effects like switching from `/` to `/posts`, or from `/posts` to `/posts_comments`, or from `/posts` to `/posts/123`, which should trigger a form reset but won't...

Can't you be a bit more specific with your test?

Reply:
Why reseting the `record-form` in those cases?

- from `/` to `/posts` (there are no `record-form` on any)
- from `/posts` to `/posts/123` (there are no `record-form` on the first and coming from any other page which may have contained one will triggered a reset)
- from `/posts` to `/posts/123` (same)